### PR TITLE
Prepare for new testnets

### PR DESCRIPTION
**Pull request by hellwolf**
- [x] New resolver using 2470 method
  - 0xce0042B868300000d44A59004Da54A005ffdcf9f
  - salt?
- [x] Refactor truffle config: network alias and usage of HDWalletProvider
- [x] ~~Missing events~~

### CODE REVIEW

File: packages/ethereum-contracts/truffle-config.js [hunk 1]
```diff
@@ -27,31 +27,66 @@ try {
...
+}
+
+/**
+ * Create default network configurations
+ */
+function createNetworkDefaultConfiguration(networkName, chainId) {
+    return {
+        provider: () =>
+            new HDWalletProvider({
+                mnemonic: getEnvValue(networkName, "MNEMONIC"),
```
Comment (d10r) [inline packages/ethereum-contracts/truffle-config.js:81]: Note that this change breaks configs using private keys with truffle.  
The legacy initialization method accepted both passphrase or private key as first argument. With the new way, the field `privateKeys` needs to be set for PKs.

Comment (hellwolf) [reply packages/ethereum-contracts/truffle-config.js:81]: are you using the privateKey initialization method?

can yhou show an example?

**Comment by hellwolf** [comment]
Related to deterministic resolver address for future deployment: to save Resolver build artifacts for future:

cat build/contracts/Resolver.json |  jq -cr '{ abi: .abi, bytecode: .bytecode}'

File: packages/ethereum-contracts/truffle-config.js (hunk 1 above)
Comment (d10r) [reply packages/ethereum-contracts/truffle-config.js:81]: Currently I just assign the PK to the mnemonic var, e.g. `XDAI_MNEMONIC=123...`.
That works with the [legacy way of initializing HDWalletProvider](https://github.com/trufflesuite/truffle/tree/develop/packages/hdwallet-provider#using-the-legacy-interface-deprecated), bcs the first argument could be mnemonic, pk, or pk array.

**Comment by hellwolf** [comment]
Current blocker for optimism: https://github.com/ethereum-optimism/optimism/pull/1720

**Comment by hellwolf** [comment]
- Create new Resolvers for new testnets and mainnets
- Change existing resolvers' admin to ltd admin
- Deploy framework and test tokens to new testnets

**Pull request merged**